### DISCUSSION TITLE
eKuiper plugin development tutorial is too vague to help the newbie

### DISCUSSION THREAD

**Discussion by huangwei2913** [lf-edge/ekuiper#1642] [General]
**Please describe your problem in detail, if necessary, you can upload the log file through the attachment**:
I have thoroughly read the link https://ekuiper.org/docs/en/latest/extension/native/develop/plugins_tutorial.html#overview
but I got confused about the structure of the plugin directory. I have some questions as follows.
can anyone describe what files should be included in the intended directory? 
What are the functions of these files? Are all the files necessary for the plugin deployment?
What things have to be done in the deployment progress? 
Many links are already invalid, please correct them

**Comment by ngjaying** [comment]
@superrxan How about we make a video

**Reply by huangwei2913** [reply]
what time your time free

**Reply by ngjaying** [reply]
We have wechat group and slack, check here https://ekuiper.org/docs/en/latest/#get-help

**Reply by huangwei2913** [reply]
sorry for the impolite. I find the question cannot be retracted.  I am very interested in the project. Thanks you for the help

**Reply by huangwei2913** [reply]
我已经申请加入了Wechat群，能否帮忙通过下，谢谢

**Comment by ngjaying** [comment]
The tutorial has updated, hopefully it is more clear now.
